Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -119,6 +119,7 @@ public RubberduckParserState(IVBE vbe, IProjectsRepository projectRepository, ID
             _projectRepository = projectRepository;
             _declarationFinderFactory = declarationFinderFactory;
             _vbeEvents = vbeEvents;
+            _queuedRequestors = new List<object>();
```

Comment:
Please use a thread-safe collection.

Reply:
Updated to use `ConcurrentStack` since at the moment we are only using the last requestor and firing the parse request only once and updated the related method to use thread-safe methods on the `ConcurrentStack`. 

PS: we could have just stored a single reference but I'm not sure if there might be any plan where we might want to notify all requestors or something in future.